### PR TITLE
Document MSTEST0037 analyzer

### PR DESCRIPTION
**Pull request by Youssef1313**
Product PR: https://github.com/microsoft/testfx/pull/4052 (merged already)

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/testing/mstest-analyzers/mstest0037.md](https://github.com/dotnet/docs/blob/0f2548e7b690f80fdb2759d3ed7778e82b84b072/docs/core/testing/mstest-analyzers/mstest0037.md) | [MSTEST0037: Use proper 'Assert' methods](https://review.learn.microsoft.com/en-us/dotnet/core/testing/mstest-analyzers/mstest0037?branch=pr-en-us-43601) |
| [docs/core/testing/mstest-analyzers/usage-rules.md](https://github.com/dotnet/docs/blob/0f2548e7b690f80fdb2759d3ed7778e82b84b072/docs/core/testing/mstest-analyzers/usage-rules.md) | [MSTest Usage rules (code analysis)](https://review.learn.microsoft.com/en-us/dotnet/core/testing/mstest-analyzers/usage-rules?branch=pr-en-us-43601) |
| [docs/navigate/devops-testing/toc.yml](https://github.com/dotnet/docs/blob/0f2548e7b690f80fdb2759d3ed7778e82b84b072/docs/navigate/devops-testing/toc.yml) | [docs/navigate/devops-testing/toc](https://review.learn.microsoft.com/en-us/dotnet/navigate/devops-testing/toc?branch=pr-en-us-43601) |

### CODE REVIEW

File: docs/core/testing/mstest-analyzers/mstest0037.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+# MSTEST0037: Use proper 'Assert' methods
+
+| Property                            | Value                                                                  |
+|-------------------------------------|------------------------------------------------------------------------|
+| **Rule ID**                         | MSTEST0037                                                             |
+| **Title**                           | Use proper 'Assert' methods                                            |
+| **Category**                        | Usage                                                                  |
+| **Fix is breaking or non-breaking** | Non-breaking                                                           |
+| **Enabled by default**              | No                                                                     |
+| **Default severity**                | Info                                                                   |
```
Comment (Youssef1313) [inline docs/core/testing/mstest-analyzers/mstest0037.md:23]: May change depending on the discussion outcome in https://github.com/microsoft/testfx/pull/4052/files#r1845291274

File: docs/core/testing/mstest-analyzers/mstest0037.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+In many cases, the better alternative provides better messages when they fail and are also easier to read.
+
+## How to fix violations
+
+Use the better alternative method.
+
+## When to suppress warnings
+
+You usually don't want to suppress warnings from this rule if you decided to opt-in for it.
```
Comment (Youssef1313) [inline docs/core/testing/mstest-analyzers/mstest0037.md:59]: If we change the analyzer to become enabled by default, then it's no longer opt-in and will need to adjust the wording here.

File: docs/core/testing/mstest-analyzers/mstest0037.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+| Property                            | Value                                                                  |
+|-------------------------------------|------------------------------------------------------------------------|
+| **Rule ID**                         | MSTEST0037                                                             |
+| **Title**                           | Use proper 'Assert' methods                                            |
+| **Category**                        | Usage                                                                  |
+| **Fix is breaking or non-breaking** | Non-breaking                                                           |
+| **Enabled by default**              | Yes                                                                    |
+| **Default severity**                | Info                                                                   |
+| **Introduced in version**           | 3.7.0                                                                  |
+| **There is a code fix**             | No                                                                     |
```
Comment (gewarren) [inline docs/core/testing/mstest-analyzers/mstest0037.md:25]:
```suggestion
| **Is there a code fix**             | No                                                                     |
```

Comment (gewarren) [reply docs/core/testing/mstest-analyzers/mstest0037.md:25]: "There is a code fix" could be misread by someone as that yes, there's a code fix.

File: docs/core/testing/mstest-analyzers/mstest0037.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+- The use of `Assert.AreEqual(true, <expression>)` or `Assert.AreEqual(false, <expression>)`.
+
+    Using `Assert.IsTrue(<expression>)` or `Assert.IsFalse(<expression>)` is a better alternative.
+
+- The use of `Assert.AreEqual(null, <expression>)` or `Assert.AreNotEqual(null, <expression>)`.
+
+    Using `Assert.IsNull(<expression>)` or `Assert.IsNotNull<expression>` is a better alternative.
+
+In many cases, the better alternative provides better messages when they fail and are also easier to read.
```
Comment (gewarren) [inline docs/core/testing/mstest-analyzers/mstest0037.md:51]:
```suggestion
In many cases, the better alternatives provide better messages when they fail and are also easier to read.
```

**Review by gewarren — APPROVED** [review]
Thanks

File: docs/core/testing/mstest-analyzers/mstest0037.md (hunk 3 above)
Comment (Youssef1313) [reply docs/core/testing/mstest-analyzers/mstest0037.md:25]: This is following the same wording as all other MSTESTxxxx docs.

https://github.com/search?q=repo%3Adotnet%2Fdocs+%22There+is+a+code+fix%22&type=code

I think it's best to address it separately for all doc pages.

**Comment by Youssef1313** [comment]
@gewarren Is this ready to merge?

File: docs/core/testing/mstest-analyzers/mstest0037.md (hunk 3 above)
Comment (Youssef1313) [reply docs/core/testing/mstest-analyzers/mstest0037.md:25]: https://github.com/dotnet/docs/issues/43785

**Pull request merged**